src/hooks/login/useWalletConnectV2Login.ts

```diff
@@ -183,55 +206,76 @@ export const useWalletConnectV2Login = ({
 
     try {
       const { approval } = await providerRef.current?.connect({
-        topic: pairing.topic
+        topic: pairing.topic,
+        events: dappEvents
       });
       if (token) {
         dispatch(setTokenLogin({ loginToken: token }));
       }
 
       try {
-        await providerRef.current?.login({ approval });
+        await providerRef.current?.login({ approval, token });
       } catch (err) {
-        console.warn('User rejected existing connection proposal ', err);
+        console.warn(WalletConnectV2Error.userRejectedExisting, err);
 
         await initiateLogin();
       }
     } catch (err) {
-      console.error('Error while attempting to retore connection ', err);
+      console.error(WalletConnectV2Error.sessionExpired, err);
       setError(WalletConnectV2Error.sessionExpired);
     } finally {
       setWcPairings(providerRef.current?.pairings);
     }
   }
 
+  async function removeExistingPairing(topic: string) {
+    try {
+      if (topic) {
+        await providerRef.current?.logout({
+          topic
+        });
+      }
+    } catch (err) {
+      console.error(WalletConnectV2Error.errorLogout, err);
+      setError(WalletConnectV2Error.errorLogout);
+    } finally {
+      setWcPairings(providerRef.current?.pairings);
+    }
+  }
+
   async function generateWcUri() {
     if (!walletConnectV2RelayAddress || !walletConnectV2ProjectId) {
       setError(WalletConnectV2Error.invalidConfig);
       return;
     }
 
-    const { uri, approval } = await providerRef.current?.connect();
-    const hasUri = Boolean(uri);
+    try {
+      const { uri, approval } = await providerRef.current?.connect({
+        events: dappEvents
+      });
+
+      const hasUri = Boolean(uri);
 
-    if (!hasUri) {
-      return;
-    }
+      if (!hasUri) {
+        return;
+      }
+
+      console.log('uri', uri);
```

Comment:
console log forgotten

Reply:
not forgotten, left on purpose because it speeds up tests without having to scan the qr code every time, will be removed at a later date